### ISSUE TITLE
operations remove 00 prefix from addresses which leads to crashes.

### ISSUE THREAD

**Issue by mhhf** [ethereumjs/ethereumjs-vm#88]
I discovered this first in ethereumjs-vm 1.4 and could reproduce it in 2.0. Strangely this bug is system specific - it lead to a crash on Ubuntu but succeeded on OS X.

The error happens on a deployment of contract A which deploys another contract B. Code was compiled with solidity 0.4.3

At some Point the stack look like the following:
```
0 0x00b68db15676a4024a0b2cfe93c41c10c2323c44  <- B's address after deployment
1 0x01
...
```
Here the operation MUL is performed which produces the stack:
```
0 0xb68db15676a4024a0b2cfe93c41c10c2323c44
...
```
Note that the new address misses its 00 prefix which later leads to an error.

**Comment by axic** [comment]
Can you tell what is the exact crash you see?

It is a bit messy how the stack items are handled, there was an unfinished PR to sort that out (#37).

**Comment by mhhf** [comment]
The exact crash is: ` Error: VM Exception while executing transaction: out of gas`

However, this is because at some point: EXTCODESIZE is called on `0xb68db15676a4024a0b2cfe93c41c10c2323c44` which ultimately leads to a empty trie lookup and with this to an invalid jump

**Comment by axic** [comment]
@mhhf thanks! Will check it soon.

**Comment by mhhf** [comment]
when can we expect a version bump with this?

**Comment by axic** [comment]
After #84 is merged, hopefully all this week.

**Comment by axic** [comment]
@mhhf 2.0.1 is released 🎉

**Comment by ethernomad** [comment]
Should this fix be backported to 1.4?

**Comment by axic** [comment]
@ethernomad the 2.x release was made way too early and as such doesn't include many changes, but this also means moving to 2.x is very simple. The only change is to update the parameter passing to the VM constructor.
